src/resolve-host/resolve-host.c

```diff
@@ -64,10 +64,12 @@ static void print_source(uint64_t flags, usec_t rtt) {
         fputs("\n-- Information acquired via", stdout);
 
         if (flags != 0)
-                printf(" protocol%s%s%s",
+                printf(" protocol%s%s%s%s%s",
```

Comment:
puts() and fputs(..., stdout) are actually quite different... The former adds a trailing newline. Which makes it pretty useless in many cases.

printf() is quite slow, which is the only reason we try to avoid it if we have trivially easy strings to output.